vault/router.go

```diff
@@ -201,14 +200,46 @@ func (r *Router) MatchingMountByAccessor(mountAccessor string) *MountEntry {
 // MatchingMount returns the mount prefix that would be used for a path
 func (r *Router) MatchingMount(path string) string {
 	r.l.RLock()
-	mount, _, ok := r.root.LongestPrefix(path)
+	var mount = r.matchingMountInternal(path)
```

Comment:
[Nitpick][Optional] We could do a `defer r.l.RUnlock()` and  `return r.matchingMountInternal(path)` and avoid the local var `mount`.